### PR TITLE
[int64] fix up

### PR DESCRIPTION
**Pull request by dorafmon**

### CODE REVIEW

**Comment by dorafmon** [comment]
@glennsl I have come to a problem when I try to use the `Comparison` module. 

I tried to type the `Container.Int64.compare` to be `t -> t -> Comparision.comparision` (on a side note I'd like it to be `Comparision.t`). And then I had a situation that `Container.Map.Make` functor expects a module `Map.OrderedType` with `compare:t->t->int`. I tried to make a functor to convert between `Comparision` and `Map.OrderedType` but it seems it does not work. Do you have any suggestions? Thanks!

**Comment by glennsl** [comment]
I think it would have to be a functor. Why didn't that work?

It's `Comparison.comparison` to be consistent with `Result.result` btw, but I don't feel strongly about keeping that.

File: src/bint64.ml [hunk 1]
```diff
@@ -1,55 +1,73 @@
...
+let maxInt = Int64.max_int
+
+let minInt = Int64.min_int
+
+let abs = Int64.abs
 
 (** {2 Conversion} *)
 
-let of_int_exn = of_int
+let ofIntExn = Int64.of_int
```
Comment (glennsl) [inline src/bint64.ml]: Should be `fromIntOrRaise`. There's quite a few other `of*` functions that should be `from*`, and a few other instances of `*Exn` that should be `*OrRaise`.

File: src/comparison.ml [hunk 2]
```diff
@@ -7,4 +7,16 @@ let compare a b = match Pervasives.compare a b with
...
+  | _ -> assert false
+
+let fromInt i = if i > 0 
+  then Greater
+  else (
+    if i = 0 then Equal
+    else Less
+  ) 
+
+let toInt i = match i with
```
Comment (glennsl) [inline src/comparison.ml]: This can be shortened a tiny bit by using the function pattern match short form

File: src/comparison.ml [hunk 3]
```diff
@@ -7,4 +7,16 @@ let compare a b = match Pervasives.compare a b with
   | 0 -> Equal
   | -1 -> Less
   | 1 -> Greater
-  | _ -> assert false
\ No newline at end of file
+  | _ -> assert false
+
+let fromInt i = if i > 0
```
Comment (glennsl) [inline src/comparison.ml]: I don't think ti's a good idea to start an if expression at the end of the line like this. It should be on the same indent level as the branches.

**Comment by dorafmon** [comment]
@glennsl Hi I fixed the code per the comment, I will merge this in now.

After some thoughts and consideration I think we should move `comparision.ml` to `bord.ml` which represents a total ordering relationship and provide utilities for that. And then change `compare` functions in modules to use `Bord.t`. And then I will get back to change `Bmap`. Its key should be a instance of `Bord.S` which represents a set with a total order relationship between elements (it is not there yet).

**Comment by glennsl** [comment]
Ok, sounds good to me!

**Pull request merged**